### PR TITLE
WIP: Attempt to speed up the Travis build

### PR DESCRIPTION
**Pull request by jwiggins**
Opening a PR to trigger Travis...

### CODE REVIEW

**Comment by jwiggins** [comment]
:boom: Yeah, that failed... Lets try something else....

**Comment by jwiggins** [comment]
So now that it's working, I'm going to break it again...

File: .travis.yml [hunk 1]
```diff
@@ -35,7 +28,9 @@ python:
 - 3.4
 
 before_install:
-    - sudo apt-get update; true
+    - ccache -s
+    - pip install --upgrade pip
+    - export PATH=/usr/lib/ccache:${PATH}
```
Comment (itziakos) [inline .travis.yml]: I checked the travis-ci documentation we do not need to do this http://docs.travis-ci.com/user/caching/#ccache-cache

Comment (itziakos) [inline .travis.yml]: We should also probably set `export CC="ccache gcc"`

**Comment by jwiggins** [comment]
So the main test script was broken up into a series of scripts to gather more granular timing information. @itziakos and I spent a couple hours looking at it this afternoon. He offered a couple recommendations (feel free to chime in if I misunderstood, Ioanni):
- Move the optional packages into their own requirements file and use the environment markers described here: https://pip.pypa.io/en/stable/reference/pip_install.html#requirement-specifiers. This will remove some of the conditionals encoded in `travis_script.sh`.
- Add more builds to the travis configuration so that independent steps can run in parallel. Namely, the normal tests, optional package tests, and doc tests are the three biggest contributors to running time.

I'll be looking at this during the EuSciPy sprints tomorrow

**Pull request closed without merging**